app/main.py

```diff
@@ -1,3 +1,46 @@
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    @staticmethod
+    def isinstance_(other: int | float) -> int | float:
+        return other.km if isinstance(other, Distance) else other
+
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float) -> Any:
+        return Distance(self.km + Distance.isinstance_(other))
+
+    def __iadd__(self, other: int | float) -> Any:
+        self.km += Distance.isinstance_(other)
+        return self
+
+    def __mul__(self, other: int | float) -> Any:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: int | float) -> Any:
+        if other != 0:
```

Comment:
Rewrite this condition to take the boolean value into an account
```suggestion
        if other:
```